include/z64camera.h

```diff
@@ -320,8 +387,1188 @@ typedef struct {
     /* 0x8 */ CameraMode* cameraModes;
 } CameraSetting; // size = 0xC
 
+/*================================
+ *   MISC DATA
+ *================================
+ */
+
+// Camera Collision Check
+typedef struct {
+    /* 0x00 */ Vec3f pos;
+    /* 0x0C */ Vec3f norm;
+    /* 0x18 */ CollisionPoly* poly;
+    /* 0x1C */ VecSph sphNorm;
+    /* 0x24 */ s32 bgId;
+} CameraCollision; // size = 0x28
+
+typedef struct {
+    /* 0x00 */ Vec3f collisionClosePoint;
+    /* 0x0C */ CameraCollision atEyeColChk;
+    /* 0x34 */ CameraCollision eyeAtColChk;
+    /* 0x5C */ f32 swingUpdateRate;
+    /* 0x60 */ s16 pitch;
+    /* 0x62 */ s16 yaw;
+    /* 0x64 */ s16 unk_64;
+    /* 0x66 */ s16 timer; // startSwingTimer
+} SwingAnimation; // size = 0x68
+
+
+/*================================
+ *   Camera Update-Function Data
+ *================================
+ */
+
+#define FLAGS_FIXED_DATA(interfaceFlags) \
+    { interfaceFlags, CAM_DATA_INTERFACE_FLAGS }
+
+
+/*================================
+ *   Camera_Normal1() Data
+ *================================
+ */
+
+#define NORMAL1_FIXED_DATA(yOffset, data01, data02, pitchTarget, eyeStepScale, posStepScale, yawDiffRange, fov, data08, interfaceFlags) \
```

Comment:
"fixed data" reminds me of the discussions we had that led to going with roData/rwData, is this intended or do you mean to use `CAM_FUNCDATA_NORMAL1` like currently in oot?

Reply:
Oh, good catch. Yeah, this is still on an old version of a name and hasn't been updated. We've moved away from "fixed" data terminology.

There is a WIP PR open for camera, https://github.com/zeldaret/mm/pull/601, and while MM's camera has it's differences, these cam flags are structured identical to OoT